clang/lib/Sema/Sema.cpp

```diff
@@ -1679,6 +1679,15 @@ class DeferredDiagnosticsEmitter
   }
 
   void visitUsedDecl(SourceLocation Loc, Decl *D) {
+    if (S.LangOpts.SYCLIsDevice && ShouldEmitRootNode) {
+      if (auto *VD = dyn_cast<VarDecl>(D)) {
```

Comment:
```suggestion
      if (const auto *VD = dyn_cast<VarDecl>(D)) {
```

Reply:
It seems I can't do that without changing signature of `checkAllowedSYCLInitializer` to make it accept `const VarDecl`. Should I do that In this PR or it will be better to do it in a separate patch?

Reply:
>>it will be better to do it in a separate patch?

Sure.